pkg/resource/deploy/step_generator.go

```diff
@@ -72,10 +72,34 @@ func (sg *stepGenerator) isTargetedUpdate() bool {
 	return sg.updateTargetsOpt != nil || sg.replaceTargetsOpt != nil
 }
 
+// isTargetedForUpdate checks if a urn is target for update.
+//
+// Note: to accommodate the `--target-dependents` flag,
+// `isTargetedForUpdateOrAddIfDep` should probably be called instead.
 func (sg *stepGenerator) isTargetedForUpdate(urn resource.URN) bool {
 	return sg.updateTargetsOpt == nil || sg.updateTargetsOpt[urn]
 }
 
+// isTargetedForUpdateOrAddIfDep checks if a target should be updated, updating
+// `sg.isTargetedForUpdateOrAddIfDep` as necessary for `--target-dependents` to be
+// accurate.
+func (sg *stepGenerator) isTargetedForUpdateOrAddIfDep(urn resource.URN, dependencies []resource.URN) bool {
+	if sg.isTargetedForUpdate(urn) {
+		return true
+	}
+	// TODO; does sg.opts.TrustDependencies play a role here?
```

Comment:
I think `TrustDependencies` is about parallelization, but I'm not sure. I don't think we need to check `TrustDependencies`.

Reply:
`TrustDependents` is about ordering--basically, we used to have Python SDKs that didn't report correct dependency info, so we used that flag to work around that. You shouldn't need to check it here, since this is more of a structural thing.